### PR TITLE
Himanshu/home-page Set up a new home page

### PR DESCRIPTION
**Pull request by HGoyal09**
Set up a new home page with stock filter 

image

image

### CODE REVIEW

File: src/components/StockTable/StockInfoTable.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    >
+        <Thead>
+            <Tr>
+                <Th>Ticker Id</Th>
+                <Th>Dividend Payout Ratio</Th>
+                <Th display={{base: 'none', md: 'table-cell'}}>PE Ratio</Th>
+                <Th display={{base: 'none', md: 'table-cell'}}>PEG Ratio</Th>
+                <Th display={{base: 'none', lg: 'table-cell'}}>Quarterly Earning Growth</Th>
+                <Th display={{base: 'none', lg: 'table-cell'}}>Quarterly Revenue Growth</Th>
+                <Th>Fifty-two Week Low</Th>
```
Comment (weiyoung) [inline src/components/StockTable/StockInfoTable.jsx]: change this and stock-details page's "52-week high" label both to the same name to keep it consistent (maybe use an enum?)

File: server/routes/stock-details.js [hunk 2]
```diff
@@ -6,6 +6,33 @@ const defaultData = {
     stock_details: {}
 }
 
+/* GET all quarterly stock details. */
+router.get('/', function (req, res) {
+    quarterlyStockInfo.find({}).then((stock_details) => {
+        const result = []
+        for (let stocks of stock_details) {
+            const {stock_details} = stocks
```
Comment (weiyoung) [inline server/routes/stock-details.js:14]: why is this variable named the same as the function argument?

File: server/routes/stock-details.js [hunk 3]
```diff
@@ -6,6 +6,33 @@ const defaultData = {
     stock_details: {}
 }
```
Comment (weiyoung) [inline server/routes/stock-details.js:8]: bring in some useful realtime data too, i.e. current price, change today, etc. can do in future commit, just a heads up :)

File: src/components/StockTable/StockSearchBar.jsx [hunk 4]
```diff
@@ -1,59 +1,19 @@
...
-                    </InputGroup>
-                </Container>
-            </HStack>
-        </Box>
-    );
-};
+        <InputGroup w='40%' borderRadius='lg' border='1px' borderColor={useColorModeValue('brand.200', 'brand.600')}>
+            <InputLeftElement
+                pointerEvents="none"
+                children={<SearchIcon color="gray.300"/>}
```
Comment (weiyoung) [inline src/components/StockTable/StockSearchBar.jsx]: don't use single gray color, use useColorModeValue and our brand color instead

File: src/components/StockTable/StockInfoTable.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import {Link, Table, Tbody, Td, Th, Thead, Tr, useColorModeValue} from "@chakra-ui/react";
+import {Link as ReactRouterLink} from "react-router-dom";
+
+const StockInfoTable = ({stocks}) => (
+    <Table variant="striped"
+           border='4px'
+           borderColor={useColorModeValue('brand.200', 'brand.600')}
```
Comment (weiyoung) [inline src/components/StockTable/StockInfoTable.jsx:8]: suggestion: remove border

File: src/components/StockTable/StockSearchBar.jsx [hunk 6]
```diff
@@ -1,59 +1,19 @@
...
+            />
+            <Input placeholder="Stock Ticker" onChange={handleChange} value={filterKeyword}/>
+            <InputRightElement children={<CloseIcon color="gray.300" onClick={handleCancel}/>}/>
```
Comment (weiyoung) [inline src/components/StockTable/StockSearchBar.jsx]: same for this CloseIcon's color

File: server/routes/stock-details.js [hunk 7]
```diff
@@ -28,4 +55,32 @@ router.get('/quarterly-data/:ticker', function (req, res) {
...
+                peg_ratio: stock_details.peg_ratio,
+                quarterly_earning_growth: stock_details.quarterly_earning_growth,
+                quarterly_revenue_growth: stock_details.quarterly_revenue_growth,
+                fifty_two_week_high: stock_details.fifty_two_week_high,
+                fifty_two_week_low: stock_details.fifty_two_week_low,
+                dividend_payout_ratio: stock_details.dividend_payout_ratio
+            })
+        }
+        res.send(result)
+    }).catch(({message}) => {
```
Comment (weiyoung) [inline server/routes/stock-details.js:80]: refactor to err =>

File: server/routes/stock-details.js [hunk 8]
```diff
@@ -6,6 +6,33 @@ const defaultData = {
     stock_details: {}
```
Comment (weiyoung) [inline server/routes/stock-details.js:6]: can we delete defaultData?

File: src/pages/Home.jsx [hunk 9]
```diff
@@ -1,24 +1,36 @@
...
+        setFilterKeyword('')
+        filterStocks('')
+    }
 
     return (
         <PageTemplate>
-            <CustomHeading>Stock Market Overview</CustomHeading>
-            <StockSearchBar />
-            <StockTableCard />
+            <VStack spacing={16} m={4}>
```
Comment (weiyoung) [inline src/pages/Home.jsx:27]: change VStack to CustomBox component that I made? and bring CustomHeading out

**Review by weiyoung — APPROVED** [review]

File: server/routes/stock-details.js (hunk 8 above)
Comment (HGoyal09) [reply server/routes/stock-details.js:6]: I would say its better to have it there to prevent duplicate code

File: src/pages/Home.jsx (hunk 9 above)
Comment (HGoyal09) [reply src/pages/Home.jsx:27]: We dnt need a box here 
We need a Vstack

**Pull request merged**